CONFIG_THREAD_LOCAL_STORAGE=y build fails with ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb

ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb is one of the supported toolchains to build Zephyr (for ARM architecture). After https://github.com/zephyrproject-rtos/zephyr/pull/28982 was merged, and with CONFIG_THREAD_LOCAL_STORAGE enabled (which it is, for one sanitycheck configuration of tests/kernel/common), the build fails with:

```
[ 87%] Linking C executable app_smem_unaligned_prebuilt.elf
/home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: /home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/thumb/v7e-m/nofp/libgcc.a(emutls.o): in function `emutls_alloc':
emutls.c:(.text+0x5a): undefined reference to `abort'
/home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: /home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/thumb/v7e-m/nofp/libgcc.a(emutls.o): in function `__emutls_get_address':
emutls.c:(.text+0x7a): undefined reference to `abort'
collect2: error: ld returned 1 exit status
make[2]: *** [zephyr/app_smem_unaligned_prebuilt.elf] Error 1
make[1]: *** [zephyr/CMakeFiles/app_smem_unaligned_prebuilt.dir/all] Error 2
make: *** [all] Error 2

INFO    - /home/buildslave/srv/zephyr/master/gnuarmemb/frdm_k64f/frdm_k64f/tests/kernel/common/kernel.common.tls/build.log
```

Full build log: https://ci.linaro.org/job/zephyr-upstream/PLATFORM=frdm_k64f,ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb,label=docker-xenial-amd64-13/5680/console

--- a/lib/libc/minimal/CMakeLists.txt
+++ b/lib/libc/minimal/CMakeLists.txt
@@ -4,6 +4,7 @@ zephyr_system_include_directories(include)
 
 zephyr_library()
 zephyr_library_sources(
+  source/stdlib/abort.c
   source/stdlib/atoi.c
   source/stdlib/strtol.c
   source/stdlib/strtoul.c

--- a/lib/libc/minimal/include/stdlib.h
+++ b/lib/libc/minimal/include/stdlib.h
@@ -35,6 +35,7 @@ static inline void exit(int status)
 {
 	_exit(status);
 }
+void abort(void);
 
 int rand(void);
 

--- a/lib/libc/minimal/source/stdlib/abort.c
+++ b/lib/libc/minimal/source/stdlib/abort.c
@@ -1,0 +1,14 @@
+/*
+ * Copyright (c) 2020 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <zephyr.h>
+
+void abort(void)
+{
+	printk("abort()\n");
+	k_panic();
+}

